src/Diagnostics.DataProviders/GeoMasterUtility.cs

```diff
@@ -0,0 +1,161 @@
+
+                if (!string.Equals(slotName, GeoMasterConstants.ProductionSlot, StringComparison.OrdinalIgnoreCase))
+                {
+                    path += GeoMasterConstants.SlotsResource + GeoMasterConstants.SlotNameTemplateParameter.NamedFormat(new Dictionary<string, string>
+                    {
+                        {"slot", slotName}
+                    });
+                }
+            }
+
+            return path;
+        }
+
+        public static string GetResourceRootPath(string subscriptionId, string resourceGroupName)
+        {
+            return GeoMasterConstants.ResourceRoot.NamedFormat(new Dictionary<string, string>
+            {
+                {"subscriptionId", subscriptionId},
+                {"resourceGroupName", resourceGroupName}
+            });
+        }
+        public static string CsmAnnotateQueryString(string queryString, string apiVersion)
+        {
+            var nameValues = HttpUtility.ParseQueryString(queryString);
+            var dict = nameValues.Cast<string>().ToDictionary(k => k, k => nameValues[k]);
+            dict["api-version"] = apiVersion;
+
+            return "?" + string.Join("&", from kvp in dict
+                                          where kvp.Value != null
+                                          select string.Format("{0}={1}", HttpUtility.UrlEncode(kvp.Key), HttpUtility.UrlEncode(kvp.Value)));
+
+
+        }
+    }
+
+    public static class SiteNameParser
+    {
+        private static readonly Regex SiteWithSlotNameRegexp =
+            new Regex(@"^(?<siteName>[^\(]+)\((?<slotName>[^\)]+)\)$");
+        private const string FmtSiteWithSlotName = "{0}({1})";
+        private const string FmtHostPrefixWithSlotName = "{0}-{1}";
+        private const string FmtHostname = "{0}.{1}";
+
+
+        public static void ParseSiteWithSlotName(string siteWithSlotName, out string siteName, out string slotName)
+        {
+            var match = SiteWithSlotNameRegexp.Match(siteWithSlotName);
+            if (match.Success)
+            {
+                siteName = match.Groups["siteName"].Value;
+                slotName = match.Groups["slotName"].Value;
+            }
+            else
+            {
+                siteName = siteWithSlotName;
+                slotName = GeoMasterConstants.ProductionSlot;
+            }
+        }
+
+        public static string GenerateSiteWithSlotName(string siteName, string slotName)
+        {
+            if (!string.IsNullOrEmpty(slotName) && !string.Equals(slotName, GeoMasterConstants.ProductionSlot, StringComparison.OrdinalIgnoreCase))
+            {
+                return string.Format(FmtSiteWithSlotName, siteName, slotName);
+            }
+
+            return siteName;
+        }
+
+        public static string GenerateHostPrefixWithSlotName(string siteName, string slotName)
+        {
+            if (!string.Equals(slotName, GeoMasterConstants.ProductionSlot, StringComparison.OrdinalIgnoreCase))
```

Comment:
what if slot name is empty? Don't know exactly how you are using this.

Reply:
All that I have is exact copy of existing code in the GeoMasterClient...

Reply:
So it looks like this will work now but only because it is currently only called with the output of ParseSiteWithSlotName, which returns 'Production'.  It is probably work checking for an empty slot name, in case someone uses it in the future without passing 'Production' when there is no slot name

Reply:
For now, leaving it as is. We can revisit this if required